### ISSUE TITLE
Various issues with miter limits

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#9830]
Noticed this a while ago (while implementing mplcairo) but prompted to report due to #9829:

```
from matplotlib import pyplot as plt, ticker

fig, axs = plt.subplots(4, 4)
for i, ax in enumerate(axs.flat):
    ax.xaxis.set_major_locator(ticker.NullLocator())
    ax.yaxis.set_major_locator(ticker.NullLocator())
    ax.plot([-1, 0, 1], [0, 1, 0], lw=i, solid_joinstyle="miter")
    ax.set(xlim=(-15, 15), ylim=(0, 3))

fig.savefig("/tmp/test.png")
fig.savefig("/tmp/test.svg")
fig.savefig("/tmp/test.pdf")

plt.show()
```

pdf appears to respect the miter ("pointed") joinstyle (funnily this appears to behave in the opposite manner as #9829):
![test-pdf](https://user-images.githubusercontent.com/1322974/33109294-9f8fccd8-cef5-11e7-9f3e-4563f903d8d5.png)
svg appears to ignore it:
![test-svg](https://user-images.githubusercontent.com/1322974/33109293-9bcf73e6-cef5-11e7-9a30-a9327cde3eac.png)
agg changes its mind half-way:
![test](https://user-images.githubusercontent.com/1322974/33109296-a2d9cb78-cef5-11e7-8bda-a42809d44767.png)

The behavior of agg is almost certainly due to a misunderstanding in how the miter limit is set.  See https://www.w3schools.com/tags/canvas_miterlimit.asp and https://www.cairographics.org/manual/cairo-cairo-t.html#cairo-set-miter-limit for the definition of the miter limit.  Note that the two definitions differ: html defines it in absolute lengths whereas cairo defines it relative to the linewidth.

backend_agg.h has the line (multiple times)
```
stroke.miter_limit(points_to_pixels(gc.linewidth));
```
which suggest that the author thought miter_limit takes an absolute miter limit, but in fact a relative miter limit should have been given (how do I know this? because I can reproduce the same behavior with mplcairo with the same setting, and cairo clearly documents that it uses a relative miter limit :-)).

So the current implementation ends up switching from miter join to bevel join at an angle that depends on the linewidth, eek.

**Comment by brunobeltran** [comment]
there are two issues here: 

1) The miter_limit is not explicitly set by matplotlib anywhere. Therefore, each backend will end up effectively utilizing whatever the target is (e.g. [4 for SVG](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-miterlimit), [10 for PostScript](https://www.ncarg.ucar.edu/gks/ps.html) [and PDF](https://help.adobe.com/pdfl_sdk/18/PDFL_SDK_HTMLHelp/PDFL_SDK_HTMLHelp/API_References/PDFL_API_Reference/PDFEdit_Layer/General.html), etc.)
2) Agg in particular appears to still treat miter limits incorrectly (I can reproduce this issue on 3.1.3).

So to really deal with this issue, it looks like we'd have to actually add a new rcParam (miterlimit), and have each of the backends respect it (including a simple change in `_backend_agg.h` to force the Agg backend to treat miter limit as relative, as that is the standard *everywhere* else).

This bug seems reasonably low priority and high effort? But if I want to correctly handle miter limits in #16607, I would need to add this new rcParam first...

**Comment by anntzer** [comment]
I agree that this is pretty high effort/low priority (though see https://github.com/matplotlib/matplotlib/issues/14939)...

**Comment by anntzer** [comment]
Still true, still low priority.
